### ISSUE TITLE
Behaviour on pythonanywhere different to local

### ISSUE THREAD

**Issue by annakrystalli** [reprohack/reprohack_site#149]
I've noticed some really weird departures in behaviour between local and pythonanywhere that I just can't figure out!

## 1. Available event to associate a paper with

I have added a [function to limit the event available to associate a paper with to only upcoming events](https://github.com/reprohack/reprohack_site/blob/1c87039dfc880d9266edf5d8bcffdb1367357fd5/reprohack_hub/models.py#L209-L210). 

This works locally but not on pythonanywhere 🤷‍♀️. I've checked and the code has propagated successfully to pythonanywhere and nothing obvious in the logs about this.

##### Event list (3 upcoming events)
![image](https://user-images.githubusercontent.com/5583057/136042850-dde3413b-ea31-4325-ac23-f86eb27b8f00.png)

##### Event options on paper form (3 upcoming events)

![image](https://user-images.githubusercontent.com/5583057/136042961-3ead7ad2-0024-4e42-9249-2142c908429c.png)

##### Event list (1 upcoming event)

![image](https://user-images.githubusercontent.com/5583057/136043208-401ef1a7-b9b7-4a1a-848c-eb50104b2e2e.png)

##### Event options on paper form (2 options - 1 upcoming & 1 past)

![image](https://user-images.githubusercontent.com/5583057/136043085-be051f7f-d953-44b9-9044-06e0b9ebf51e.png)

## 2. Paper List

The paper list page should show any paper that is not archived and is available for general review. Her's the code: https://github.com/reprohack/reprohack_site/blob/1c87039dfc880d9266edf5d8bcffdb1367357fd5/reprohack_hub/models.py#L304-L305

### Local

Paper list correctly shows any papers available for general review (indicated by the available for general review icon)

![image](https://user-images.githubusercontent.com/5583057/136043976-c3375cfb-cd6e-4375-b1cb-415e0e871971.png)

### pythonanywhere

There is a single test paper currently in the database which is set to be available for general review and not archived. It is visible on the authors profile with the correct icons indicating status:
![image](https://user-images.githubusercontent.com/5583057/136045881-96a82309-7f2e-4d1a-8ad0-900f4319257c.png)

However it is not visible in the paper list 🤷‍♀️

![image](https://user-images.githubusercontent.com/5583057/136045999-55398fe1-fb77-4fa3-8cc5-47bd4509457f.png)

### Server Logs

Here are the server logs since I merged the PR last night:
```
2021-10-04 23:24:45 ERROR 2021-10-04 23:24:45,630 user_wsgi_wrapper 5 139929205917568 /home/reprohacks/reprohack_site/.virtualenv/lib/python3.8/site-packages/django/views/generic/list.py:86: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'reprohack_hub.models.Review'> QuerySet.
2021-10-04 23:24:45 ERROR 2021-10-04 23:24:45,632 user_wsgi_wrapper 5 139929205917568   return self.paginator_class(
2021-10-05 09:33:22 Tue Oct  5 09:33:22 2021 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request / (ip 10.0.0.104) !!!
2021-10-05 09:33:22 Tue Oct  5 09:33:22 2021 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET / (10.0.0.104)
2021-10-05 09:33:22 ERROR 2021-10-05 09:33:22,791 user_wsgi_wrapper 5 139929205917568 OSError: write error
2021-10-05 09:33:23 Tue Oct  5 09:33:23 2021 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request / (ip 10.0.0.104) !!!
2021-10-05 09:33:23 Tue Oct  5 09:33:23 2021 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET / (10.0.0.104)
2021-10-05 09:33:23 ERROR 2021-10-05 09:33:23,510 user_wsgi_wrapper 5 139929205917568 OSError: write error
2021-10-05 14:14:25 ERROR 2021-10-05 14:14:25,286 user_wsgi_wrapper 8 139929205917568 /home/reprohacks/reprohack_site/.virtualenv/lib/python3.8/site-packages/django/views/generic/list.py:86: UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'reprohack_hub.models.Review'> QuerySet.
2021-10-05 14:14:25 ERROR 2021-10-05 14:14:25,296 user_wsgi_wrapper 8 139929205917568   return self.paginator_class(
```

There do seem to be some errors but they seem to be pagination errors associated with the review list page.

Any ideas @twinkarma ? I just can't figure it out!

**Comment by annakrystalli** [comment]
Aha! Thanks for that. The old turn it off and on again 😂

Re: moving filtering to the queryset. Agreed the current approach is hacky and will break pagination. Could you give me a bit of guidance on how to implement it though? I don't understand how to combine it with  the search query (sorry I still don't understand the queryset functionality that well yet).

**Comment by twinkarma** [comment]
It was literally 3 lines of code so I'd implemented for your here reprohack/reprohack_site@eaccaf908e004779025c7b6e39ed197a24c12c8b , it's included in pr #148 

(Sorry, posted on the wrong issue before)
